addons/account/tests/test_early_payment_discount.py

```diff
@@ -60,6 +60,71 @@ def test_early_payment_end_date(self):
                     fields.Date.from_string('2019-01-11') or False
                 )
 
+    # ========================== Tests Taxes Amounts =============================
+    def test_fixed_tax_amount_discounted_payment_mixed(self):
+        self.env.company.early_pay_discount_computation = 'mixed'
+        fixed_tax = self.env['account.tax'].create({
+            'name': 'Test 0.05',
+            'type_tax_use': 'sale',
+            'amount_type': 'fixed',
+            'amount': 0.05,
+            'price_include': False,
+            'include_base_amount': False,
+        })
+
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2019-01-01',
+            'date': '2019-01-01',
+            'invoice_line_ids': [
+                Command.create({'name': 'line', 'price_unit': 1000.0, 'tax_ids': [Command.set(self.product_a.taxes_id.ids + fixed_tax.ids)]}),
+            ],
+            'invoice_payment_term_id': self.early_pay_mixed_5_10.id,
+        })
+        self.assertInvoiceValues(invoice, [
+            {
+                'display_type': 'epd',
+                'balance': -75.0,
+                'price_subtotal': 0.0,
+                'price_total': 0.0,
+            }, {
+                'display_type': 'epd',
+                'balance': 75.0,
+                'price_subtotal': 0.0,
+                'price_total': 0.0,
+            }, {
+                'display_type': 'product',
+                'balance': -1000.0,
+                'price_subtotal': 1000.0,
+                'price_total': 1150.05,
+            }, {
+                'display_type': 'tax',
+                'balance': -138.75,
+                'price_subtotal': 0.0,
+                'price_total': 0.0,
+            }, {
+                'display_type': 'tax',
+                'balance': -0.05,
+                'price_subtotal': 0.0,
+                'price_total': 0.0,
+            }, {
+                'display_type': 'payment_term',
+                'balance': 569.4,
+                'price_subtotal': 0.0,
+                'price_total': 0.0,
+            }, {
+                'display_type': 'payment_term',
+                'balance': 569.4,
+                'price_subtotal': 0.0,
+                'price_total': 0.0,
+            },
+        ], {
+            'amount_untaxed': 1000.0,
+            'amount_tax': 138.8,
+            'amount_total': 1138.8,
+        })
```

Comment:
self.assertRecordValues(invoice.line_ids, [
```suggestion
    # pylint: disable=bad-whitespace
    {'display_type': 'epd',             'balance': -75.0},
    {'display_type': 'epd',             'balance': 75.0},
    {'display_type': 'product',         'balance': -1000.0},
    {'display_type': 'tax',             'balance': -138.75},
    {'display_type': 'tax',             'balance': -0.05},
    {'display_type': 'payment_term',    'balance': 569.4},
    {'display_type': 'payment_term',    'balance': 569.4},
])
self.assertRecordValues(invoice, [{
    'amount_untaxed': 1000.0,
    'amount_tax': 138.8,
    'amount_total': 1138.8,
}])
```